src/main/java/filter/expression/Qualifier.java

```diff
@@ -527,18 +538,34 @@ private boolean satisfiesUpdatedHours(TurboIssue issue) {
             return false;
         }
 
-        int hoursSinceUpdate;
+        LocalDateTime dateOfUpdate = null;
 
-        if (issue.getMetadata().isUpdated()) {
-            // Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt.
-            hoursSinceUpdate = Utility.safeLongToInt(issue.getMetadata().getNonSelfUpdatedAt()
-                    .until(getCurrentTime(), ChronoUnit.HOURS));
-        } else {
-            // First time being filtered (haven't gotten metadata from source yet).
-            hoursSinceUpdate = Utility.safeLongToInt(issue.getUpdatedAt().until(getCurrentTime(), ChronoUnit.HOURS));
+        //Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt
+        //and getSelfUpdatedAt
+        switch (updatedKind){
+            case SELF_UPDATED:
+                if (issue.getMetadata().isUpdatedBySelf()) {
+                    dateOfUpdate = issue.getMetadata().getSelfUpdatedAt();
+                }
+                break;
+            case OTHER_UPDATED:
+                if (issue.getMetadata().isUpdatedByOthers()) {
+                    dateOfUpdate = issue.getMetadata().getNonSelfUpdatedAt();
+                }
+                break;
+            case ALL_UPDATED:
+                // The or condition for dateOfUpdate is for first time check
+                if (issue.getMetadata().isUpdated() || dateOfUpdate == null) {
+                    dateOfUpdate = issue.getUpdatedAt();
+                }
         }
 
-        return updatedRange.encloses(hoursSinceUpdate);
+        if (dateOfUpdate != null){
+            int hoursSinceUpdate = Utility.safeLongToInt(dateOfUpdate.until(getCurrentTime(), ChronoUnit.HOURS));
+            return updatedRange.encloses(hoursSinceUpdate);
+        } else {
+            return false;
+        }
```

Comment:
Yup, as a rule of thumb, 'pick the lower hanging fruits first' to improve readability.